Hw1

awong-pdx opened the pull request:
### Code Changes I Made and Why
I added code to 01-selectors.md, 02-palindrome.html, and 02-palindrome.js to complete the requirements for Homework 1 - Setup and Review.

***

### What Feedback I'm Looking For
Following the homework guidelines, I would appreciate any feedback on:
 - **Accessibility**
 - **Responsiveness**
 - **Airbnb style guide rules**
 - **Unnecessary code**
 - **Proper and consistent formatting**
 
***

#### Thank you for your feedback!

awong-pdx committed:
Completed 01-selectors

Added answers to remaining levels of the
CSS Diner Game in 01-selectors.md

awong-pdx committed:
Started Exercise 02 - Palindrome

Added a handleInput function to check for a palindrome
and implemented a simple check for
positive number input.

awong-pdx committed:
Added checkPalindrome function

Implemented checkPalindrome to check any Number to
determine if it is a palindrome.

awong-pdx committed:
Added palindrome check for user input

Implemented checking user input for a palindromic number and
included a check for negative numbers or text.

awong-pdx committed:
Fixed typed input handling

Updated handleInput to check each time the user
typed or removed a digit in input

awong-pdx committed:
Added Bootstrap styling

Added Bootstrap styles following assignment doc.
Added label to fix accessibility error.

awong-pdx committed:
Fixed contrast issue - background color

The Exercise 02 background had insufficient contrast with
the success, failure, and error result text colors.
Changed the background to white.

awong-pdx committed:
Removed console log statements

Removed one unnecessary console log statement.

awong-pdx committed:
Reformatted to follow airbnb style guide

Installed eslint and changed 02-palindrome.js according to
linter errors.

HaliTran reviewed (approved):
Hi Alessandra!

In terms of accessibility, responsiveness, formatting, and following the Airbnb style guides, your code met all the requirements. I couldn't find any errors pertaining to those three.

Your implementation of checking if the number is a Palindrome works as well! Although there is no right or wrong answer, I did want to point out that JavaScript does provide methods for the string type which allows you to reverse a string. Using this approach, you can check if the reverse of the string is equal to the original string without the need for a loop.

Despite this, your implementation is correct and I would argue is probably a better approach since it can easily be translated to other languages that do not provide string methods.

I really liked how you separated the Palindrome check with the input check as it introduces loose coupling within the program and makes it easier to test and move forward (if needed). I didn't take this into consideration for my approach and I find it much more efficient.

Lastly, your responses for the CSS diners all work as intended, however, for number 19 (:last-child) and 24 (:last-of-type), it can be written more concisely by targeting the small class so that you don't need to have two different CSS selections.

Great job!

HaliTran commented on hw1/01-selectors.md:
```diff
@@ -2,6 +2,33 @@
+21. plate:nth-of-type(even)
+22. plate:nth-of-type(2n+3)
+23. plate apple:only-of-type
+24. orange:last-of-type, apple:last-of-type
```
Similarly, what is common (class, id, etc) between both orange and apple in this example that you can select?

HaliTran commented on hw1/01-selectors.md:
```diff
@@ -2,6 +2,33 @@
+14. plate > apple
+15. orange:first-child
+16. plate :only-child
+17. #fancy :last-child, pickle:last-child
```
This works just fine, however, it can be a bit long. What is common (class, id, etc) between #fancy and pickle in this example that you can select?

awong-pdx replied:
Fixed those, thanks!

awong-pdx commented:
@HaliTran thanks for reviewing my code and I appreciate all of the detailed feedback!
> Your implementation of checking if the number is a Palindrome works as well! Although there is no right or wrong answer, I did want to point out that JavaScript does provide methods for the string type which allows you to reverse a string. Using this approach, you can check if the reverse of the string is equal to the original string without the need for a loop.
>
I didn't know that was a method I could use, so thanks for pointing that out!
>however, for number 17 (:last-child) and 24 (:last-of-type), it can be written more concisely by targeting the small class so that you don't need to have two different CSS selections.
>
That's so much easier, thanks for catching that.

awong-pdx committed:
Made requested changes

Simplified two answers in 01-selectors.md and
removed an unnecessary space in 02-palindrome.js

swethasrihari commented:
Hi Alessandra,

your code looks good. It has no issues with accessibility, responsiveness, formatting and following Airbnb style.

Regarding the CSS selectors, after looking at your work, I think mine needs more changes. So no comments on that as it looks good to me.

I do agree with Hali regarding the use of string functions to reverse the entered number but your method is also correct. 

Overall you did a good job!

awong-pdx commented:
> Hi Alessandra,
> 
> your code looks good. It has no issues with accessibility, responsiveness, formatting and following Airbnb style.
> 
> Regarding the CSS selectors, after looking at your work, I think mine needs more changes. So no comments on that as it looks good to me.
> 
> I do agree with Hali regarding the use of string functions to reverse the entered number but your method is also correct.
> 
> Overall you did a good job!

Thanks for reviewing my code Swetha! I think I'll look into simplifying that palindrome function, since you both suggested there's an easier way.

awong-pdx committed:
Simplified checkPalindrome()

Used String functions instead of a loop to reverse the input and
check for equality.
